crates/viewer/re_space_view_graph/src/layout/geometry.rs

```diff
@@ -0,0 +1,76 @@
+//! Provides geometric (shape) abstractions for the different elements of a graph layout.
+
+use egui::{Pos2, Rect, Vec2};
+
+#[derive(Clone, Debug)]
+pub enum PathGeometry {
+    /// A simple straight edge.
+    Line { source: Pos2, target: Pos2 },
+
+    /// Represents a cubic bezier curve.
+    ///
+    /// In the future we could probably support more complex splines.
+    CubicBezier {
```

Comment:
Note: we already depends on `kurbo`. I have a good experience using this crate for geometric primitives. Might be worth using it for some computations (e.g. bbox, linear approximation, etc.) if the need increases.

Reply:
Awesome, I heard good things about `kurbo` too!

Reply:
I've been using it lots in my personal projects.